lua/rollingLimit.lua

```diff
@@ -52,16 +53,30 @@ local retryDelta = 0
 local rejected = false
 local forced = false
 
+-- Whether this withdrawal is greater than our limit
+local largerWithdrawal = false
+
 if netTokens < 0 then -- we used more than we have
    if force then
       forced = true
       netTokens = 0 -- drain the swamp
+   elseif allowLargerWithdrawal and grossTokens == limit then
```

Comment:
As @max-b and I discussed on video, there is a potential issue here, which is that the token bucket has to be completely full of tokens in order for oversized messages to get processed. 

This is necessary because if you let them be processed when the token bucket is half-full, or at zero or greater, then you'd open up the possibility of multiple oversized messages no different numbers being processed, which would not get slowed down by the per-number rate-limiting at Bandwidth and would therefore be more likely to hit the per-campaign rate limit at the carriers.

But this means that during bursty times, when the token bucket might often not be full, oversized messages would constantly be sent to the back of the line, and would effectively have to wait until the end of the burst.

This will probably not be a problem because 

1. The sustained bursts will probably be short, and 
2. If it's an initial message or a followup, all the messages would probably be oversized, so none would be delayed more than others. The only real issue would be if an agent made a particularly long reply to a lead; then it might get held up.

But it's something to keep in the back of our minds, in case we start recognizing it happening in production in a problematic way.